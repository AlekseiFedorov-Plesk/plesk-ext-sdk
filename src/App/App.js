// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { createElement, LocaleProvider, PropTypes } from '@plesk/ui-library';
import { BrowserRouter, Switch } from 'react-router-dom';
import { ConfigContext } from '../Config';
import StatusMessages from '../StatusMessages';
import Route from '../Route';

const App = ({ locale, routes, ...props }) => (
    <ConfigContext.Provider
        value={{
            baseUrl: props.baseUrl,
        }}
    >
        <LocaleProvider messages={locale}>
            <BrowserRouter basename={props.baseUrl}>
                <StatusMessages>
                    <Switch>
                        {routes.map(({ path, component: Tag, ...routeProps }) => (
                            <Route
                                key={path}
                                path={path}
                                render={() => <Tag {...props} />}
                                routes={routes}
                                {...routeProps}
                            />
                        ))}
                    </Switch>
                </StatusMessages>
            </BrowserRouter>
        </LocaleProvider>
    </ConfigContext.Provider>
);

App.propTypes = {
    routes: PropTypes.array.isRequired,
    locale: PropTypes.object.isRequired,
    baseUrl: PropTypes.string.isRequired,
};

export default App;
