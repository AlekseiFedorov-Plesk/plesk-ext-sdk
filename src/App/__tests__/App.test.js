// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { createElement } from '@plesk/ui-library';
import App from '../App';
import { shallow } from 'enzyme';

describe('App', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <App
                baseUrl=""
                locale={{}}
                routes={[{
                    path: '/',
                    componentPath: 'Index',
                    component: () => null,
                }]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
