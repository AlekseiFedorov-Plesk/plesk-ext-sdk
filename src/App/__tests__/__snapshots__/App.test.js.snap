// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    baseUrl=""
    locale={Object {}}
    routes={
      Array [
        Object {
          "component": [Function],
          "componentPath": "Index",
          "path": "/",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <LocaleProvider
        messages={Object {}}
      >
        <ConfigProvider
          value={
            Object {
              "get": [Function],
              "post": [Function],
            }
          }
        >
          <BrowserRouter
            basename=""
          >
            <withRouter(StatusMessages)>
              <Switch>
                <Route
                  componentPath="Index"
                  path="/"
                  render={[Function]}
                />
              </Switch>
            </withRouter(StatusMessages)>
          </BrowserRouter>
        </ConfigProvider>
      </LocaleProvider>,
      "value": Object {
        "baseUrl": "",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ConfigProvider
          value={
            Object {
              "get": [Function],
              "post": [Function],
            }
          }
        >
          <BrowserRouter
            basename=""
          >
            <withRouter(StatusMessages)>
              <Switch>
                <Route
                  componentPath="Index"
                  path="/"
                  render={[Function]}
                />
              </Switch>
            </withRouter(StatusMessages)>
          </BrowserRouter>
        </ConfigProvider>,
        "messages": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <BrowserRouter
            basename=""
          >
            <withRouter(StatusMessages)>
              <Switch>
                <Route
                  componentPath="Index"
                  path="/"
                  render={[Function]}
                />
              </Switch>
            </withRouter(StatusMessages)>
          </BrowserRouter>,
          "value": Object {
            "get": [Function],
            "post": [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "basename": "",
            "children": <withRouter(StatusMessages)>
              <Switch>
                <Route
                  componentPath="Index"
                  path="/"
                  render={[Function]}
                />
              </Switch>
            </withRouter(StatusMessages)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Switch>
                <Route
                  componentPath="Index"
                  path="/"
                  render={[Function]}
                />
              </Switch>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    componentPath="Index"
                    path="/"
                    render={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "/",
                  "nodeType": "class",
                  "props": Object {
                    "componentPath": "Index",
                    "path": "/",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <LocaleProvider
          messages={Object {}}
        >
          <ConfigProvider
            value={
              Object {
                "get": [Function],
                "post": [Function],
              }
            }
          >
            <BrowserRouter
              basename=""
            >
              <withRouter(StatusMessages)>
                <Switch>
                  <Route
                    componentPath="Index"
                    path="/"
                    render={[Function]}
                  />
                </Switch>
              </withRouter(StatusMessages)>
            </BrowserRouter>
          </ConfigProvider>
        </LocaleProvider>,
        "value": Object {
          "baseUrl": "",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ConfigProvider
            value={
              Object {
                "get": [Function],
                "post": [Function],
              }
            }
          >
            <BrowserRouter
              basename=""
            >
              <withRouter(StatusMessages)>
                <Switch>
                  <Route
                    componentPath="Index"
                    path="/"
                    render={[Function]}
                  />
                </Switch>
              </withRouter(StatusMessages)>
            </BrowserRouter>
          </ConfigProvider>,
          "messages": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <BrowserRouter
              basename=""
            >
              <withRouter(StatusMessages)>
                <Switch>
                  <Route
                    componentPath="Index"
                    path="/"
                    render={[Function]}
                  />
                </Switch>
              </withRouter(StatusMessages)>
            </BrowserRouter>,
            "value": Object {
              "get": [Function],
              "post": [Function],
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "basename": "",
              "children": <withRouter(StatusMessages)>
                <Switch>
                  <Route
                    componentPath="Index"
                    path="/"
                    render={[Function]}
                  />
                </Switch>
              </withRouter(StatusMessages)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Switch>
                  <Route
                    componentPath="Index"
                    path="/"
                    render={[Function]}
                  />
                </Switch>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Route
                      componentPath="Index"
                      path="/"
                      render={[Function]}
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "/",
                    "nodeType": "class",
                    "props": Object {
                      "componentPath": "Index",
                      "path": "/",
                      "render": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
